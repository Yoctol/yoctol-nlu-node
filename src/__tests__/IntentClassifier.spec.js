const IntentClassifier = require('../IntentClassifier');

const setup = () => {
  const id = '1';
  const graphql = jest.fn();
  const intentClassifier = new IntentClassifier({ id, graphql });

  return {
    id,
    graphql,
    intentClassifier,
  };
};

it('should be defined', () => {
  expect(IntentClassifier).toBeDefined();
});

describe('handle error', () => {
  it('should throw when not provide graphql', () => {
    const createIntentClassifier = () => {
      // eslint-disable-next-line no-new
      new IntentClassifier({ id: '1' });
    };

    expect(createIntentClassifier).toThrowErrorMatchingSnapshot();
  });

  it('should throw error when train error happended', async () => {
    const { intentClassifier, graphql } = setup();

    graphql.mockResolvedValueOnce({
      data: null,
      errors: [new Error('Fake train error message')],
    });

    try {
      await intentClassifier.train();
    } catch (err) {
      expect(err.message).toMatch(/IntentClassifier#train/);
      expect(err).toMatchSnapshot();
    }
  });

  it('should throw error when predict error happended', async () => {
    const { intentClassifier, graphql } = setup();

    graphql.mockResolvedValueOnce({
      data: null,
      errors: [new Error('Fake predict error message')],
    });

    try {
      await intentClassifier.predict();
    } catch (err) {
      expect(err.message).toMatch(/IntentClassifier#predict/);
      expect(err).toMatchSnapshot();
    }
  });
});

describe('instance', () => {
  it('should have id getter', () => {
    const { id, intentClassifier } = setup();

    expect(intentClassifier.id).toBe(id);
  });
});
