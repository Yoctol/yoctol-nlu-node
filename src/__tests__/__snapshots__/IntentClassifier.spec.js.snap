// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handle error should throw error when predict error happended 1`] = `
[Error: YNLU error happens in IntentClassifier#predict!
Fake predict error message]
`;

exports[`handle error should throw error when train error happended 1`] = `
[Error: YNLU error happens in IntentClassifier#train!
Fake train error message]
`;

exports[`handle error should throw when not provide graphql 1`] = `"Must provide graphql client for NLU service."`;
